Mathlib/Algebra/Category/Grp/LargeColimits.lean

```diff
@@ -0,0 +1,84 @@
+/-
+Copyright (c) 2025 Sophie Morel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sophie Morel
+-/
+import Mathlib.Algebra.Category.Grp.Colimits
+import Mathlib.Algebra.Module.CharacterModule
+import Mathlib.Algebra.Group.Equiv.Basic
+
+/-!
+# Existence of "big" colimits in the category of additive commutative groups
+
+If `F : J ⥤ AddCommGrp.{w}` is a functor, we show that `F` admits a colimit if and only
+if `Colimits.Quot F` (the quotient of the direct sum of the commutative groups `F.obj j`
+by the relations given by the morphisms in the diagram) is `w`-small.
+
+-/
+
+universe w u v
+
+open CategoryTheory Limits
+
+namespace AddCommGrp
+
+variable {J : Type u} [Category.{v} J] {F : J ⥤ AddCommGrp.{w}} (c : Cocone F)
+
+open Colimits
+
+/--
+If `c` is a cocone of `F` such that `Quot.desc F c` is bijective, then `c` is a colimit
+cocone of `F`.
+-/
+lemma isColimit_iff_bijective_desc [DecidableEq J] :
+     Nonempty (IsColimit c) ↔ Function.Bijective (Quot.desc F c) := by
+  refine ⟨fun ⟨hc⟩ => ?_, fun h ↦ Nonempty.intro (isColimit_of_bijective_desc F c h)⟩
+  change Function.Bijective (Quot.desc F c).toIntLinearMap
+  rw [← CharacterModule.dual_bijective_iff_bijective]
+  refine ⟨fun χ ψ eq ↦ ?_, fun χ ↦ ?_⟩
+  · dsimp at eq
+    suffices eq : ofHom (AddEquiv.ulift.symm.toAddMonoidHom.comp χ) =
+        ofHom (AddEquiv.ulift.symm.toAddMonoidHom.comp ψ) by
+      apply (AddMonoidHom.postcompEquiv (@AddEquiv.ulift (AddCircle (1 : ℚ)) _).symm _).injective
+      dsimp at eq ⊢
+      ext
+      rw [← ofHom_apply, ← ofHom_apply, eq]
+    refine hc.hom_ext (fun j ↦ ?_)
+    ext x
+    change (ofHom (AddEquiv.ulift.symm.toAddMonoidHom.comp χ)) _ =
+      (ofHom (AddEquiv.ulift.symm.toAddMonoidHom.comp ψ)) _
+    dsimp
+    simp only [EmbeddingLike.apply_eq_iff_eq]
+    erw [← Quot.ι_desc _ c j x]
+    change χ.comp (Quot.desc F c).toIntLinearMap.toAddMonoidHom _ = _
+    rw [eq]
+    dsimp
```

Comment:
If you add the following lemma in `Algebra.Category.Grp.Basic`:
```lean
lemma ofHom_injective {X Y : Type u} [AddCommGroup X] [AddCommGroup Y] :
    Function.Injective (fun (f : X →+ Y) ↦ ofHom f) := by
  intro f g h
  ext
  apply DFunLike.congr_fun h
```
This becomes:
```suggestion
  · apply (AddMonoidHom.postcompEquiv (@AddEquiv.ulift (AddCircle (1 : ℚ)) _).symm _).injective
    apply ofHom_injective
    refine hc.hom_ext (fun j ↦ ?_)
    ext x
    rw [comp_apply, comp_apply, ← Quot.ι_desc _ c j x]
    exact ULift.down_injective (DFunLike.congr_fun eq (Quot.ι F j x))
```